### PR TITLE
feat(search): Add support for Elasticsearch object field type

### PR DESCRIPTION
**Pull request by justinas-marozas**
This change will add support for object field type through `@Searchable` annotations. 

As large number of keys on an object field type would create equally large number of columns in the index - I've also added a setting `elasticsearch.index.maxObjectKeys` to limit how many keys can be indexed.

### CODE REVIEW

**Review by RyanHolstien — APPROVED** [review]
Great work, thanks for the contribution! 😄

File: metadata-io/src/test/java/com/linkedin/metadata/search/elasticsearch/indexbuilder/MappingsBuilderTest.java [hunk 1]
```diff
@@ -85,6 +85,11 @@ public void testMappingsBuilder() {
     assertTrue(nestedForeignKeySubfields.containsKey("keyword"));
     assertTrue(nestedForeignKeySubfields.containsKey("ngram"));
 
+    // OBJECT
+    Map<String, Object> esObjectField = (Map<String, Object>) properties.get("esObjectField");
```
Comment (jjoyce0510) [inline metadata-io/src/test/java/com/linkedin/metadata/search/elasticsearch/indexbuilder/MappingsBuilderTest.java:89]: Overall my major comment is that we should rename esObjectField to something else.

Can we just call it "objectField" _everywhere_?

The idea of field types is to add some abstraction between us and elastic / the underlying search service. By putting a bunch of "esX" variables in the code it kind of moves us away from that goal!

File: metadata-service/factories/src/main/resources/application.yml [hunk 2]
```diff
@@ -150,6 +150,7 @@ elasticsearch:
     numReplicas: ${ELASTICSEARCH_NUM_REPLICAS_PER_INDEX:1}
     numRetries: ${ELASTICSEARCH_INDEX_BUILDER_NUM_RETRIES:3}
     maxArrayLength: ${SEARCH_DOCUMENT_MAX_ARRAY_LENGTH:1000}
+    maxObjectKeys: ${SEARCH_DOCUMENT_MAX_OBJECT_KEYS:1000}
```
Comment (jjoyce0510) [inline metadata-service/factories/src/main/resources/application.yml:153]: Nice!

File: test-models/src/main/pegasus/com/datahub/test/TestEntityInfo.pdl [hunk 3]
```diff
@@ -75,4 +75,13 @@ record TestEntityInfo includes CustomProperties {
...
\ No newline at end of file
+
+  @Searchable = {
+    "/*": {
+      "name": "esObjectField",
+      "fieldType": "OBJECT",
+      "queryByDefault": true
+    }
+  }
+  esObjectField: optional map[string, string]
```
Comment (jjoyce0510) [inline test-models/src/main/pegasus/com/datahub/test/TestEntityInfo.pdl:86]: Minor: Can we rename these to just "objectField"

File: metadata-io/src/test/java/com/linkedin/metadata/search/elasticsearch/indexbuilder/MappingsBuilderTest.java (hunk 1 above)
Comment (justinas-marozas) [reply metadata-io/src/test/java/com/linkedin/metadata/search/elasticsearch/indexbuilder/MappingsBuilderTest.java:89]: I can rename it. My thinking was that it's a test model so I named the field after the feature I wanted to test.

**Pull request merged**